File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/autonomous/actions/WaitAction.java
package org.firstinspires.ftc.teamcode.autonomous.actions;

import com.pedropathing.util.Timer;

import org.firstinspires.ftc.teamcode.autonomous.AutonomousAction;
import org.firstinspires.ftc.teamcode.hardware.Robot;

/**
 * Action that waits for a specified duration before completing.
 */
public class WaitAction implements AutonomousAction {
    private final double durationSeconds;
    private Timer timer;

    public WaitAction(double durationSeconds) {
        this.durationSeconds = durationSeconds;
    }

    @Override
    public void initialize(Robot bot) {
        timer = new Timer();
        timer.resetTimer();
    }

    @Override
    public boolean execute(Robot bot) {
        return timer.getElapsedTimeSeconds() >= durationSeconds;
    }

    @Override
    public void end(Robot bot, boolean interrupted) {
        // Nothing to clean up
    }

    @Override
    public String getName() {
        return "Wait(" + durationSeconds + "s)";
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/kotlin_mirror/autonomous/actions/WaitAction.kt
package org.firstinspires.ftc.teamcode.kotlin_mirror.autonomous.actions

import com.pedropathing.util.Timer
import org.firstinspires.ftc.teamcode.kotlin_mirror.autonomous.AutonomousAction
import org.firstinspires.ftc.teamcode.kotlin_mirror.hardware.Robot

/**
 * An action that does nothing but wait for a specified duration.
 *
 * Useful for adding delays between actions, allowing mechanisms to settle, or
 * synchronizing with alliance partners.
 */
class WaitAction(private val durationSeconds: Double) : AutonomousAction {
    private var timer: Timer? = null

    override fun initialize(bot: Robot) {
        timer = Timer()
        timer!!.resetTimer()
    }

    override fun execute(bot: Robot): Boolean {
        return timer!!.elapsedTimeSeconds >= durationSeconds
    }

    override fun end(bot: Robot, interrupted: Boolean) {
        // No cleanup needed
    }

    override val name: String
        get() = "Wait(${durationSeconds}s)"
}
